packages/flutter/lib/src/material/drawer.dart

```diff
@@ -427,12 +434,17 @@ class DrawerControllerState extends State<DrawerController> with SingleTickerPro
   Widget _buildDrawer(BuildContext context) {
     final bool drawerIsStart = widget.alignment == DrawerAlignment.start;
     final EdgeInsets padding = MediaQuery.of(context).padding;
-    double dragAreaWidth = drawerIsStart ? padding.left : padding.right;
-
-    if (Directionality.of(context) == TextDirection.rtl)
-      dragAreaWidth = drawerIsStart ? padding.right : padding.left;
+    final TextDirection textDirection = Directionality.of(context);
+
+    double dragAreaWidth = widget.edgeDragWidth;
+    if (widget.edgeDragWidth == null) {
+      if (drawerIsStart && textDirection == TextDirection.ltr || !drawerIsStart && textDirection == TextDirection.rtl) {
```

Comment:
Although it may not make things clearer, flutter style is to use a switch for textDirection.